### PR TITLE
Embed wasm for cjs build

### PR DESCRIPTION
**Pull request by asutula**

### CODE REVIEW

File: main.js
Comment (asutula) [inline main.js:1]: This file now exists solely to build the cjs version from.

File: test/main.test.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 import { rejects, strictEqual, deepStrictEqual, match } from "assert";
 import { test, before, describe } from "mocha";
 // eslint-disable-next-line no-unused-vars
-import _init, { initSync, __wasm, init } from "../main.js";
+import _init, { init } from "../module.js";
```
Comment (asutula) [inline test/main.test.js:5]: I think the is the appropriate change, but the tests fail with `Unknown file extension ".wasm" for /Users/aaron/Code/textile/wasm-sqlparser/main.wasm`.

File: package.json [hunk 2]
```diff
@@ -2,32 +2,30 @@
...
   "type": "module",
-  "main": "./cjs/main.js",
+  "main": "./out.cjs",
   "module": "./module.js",
   "types": "types.d.ts",
   "devDependencies": {
     "@types/assert": "^1.5.6",
     "@types/mocha": "^10.0.1",
-    "@types/node": "^18.16.3",
-    "@typescript-eslint/parser": "^5.59.2",
```
Comment (asutula) [inline package.json:26]: I'm very confused, this is not an accurate diff, these lines are not deleted. Maybe some GH bug!?

File: package.json [hunk 3]
```diff
@@ -2,32 +2,30 @@
...
     "@types/node": "^20.4.0",
     "@typescript-eslint/eslint-plugin": "^5.61.0",
     "@typescript-eslint/parser": "^5.61.0",
```
Comment (asutula) [inline package.json:25]: This was weird with the duplicate deps. Cleaned it up, but let me know if there is some justification for this.

File: main.js
Comment (asutula) [inline main.js:1]: This file now exists just to build the cjs version.

**Review by carsonfarmer — APPROVED** [review]
:thu

**Comment by asutula** [comment]
This is looking pretty good. I'm not sure if this is a problem with the cjs build of the parser, or some config thing with nextjs/webpack, but I'm seeing this error now:
```
../wasm-sqlparser/cjs/out.cjs
Critical dependency: require function is used in a way in which dependencies cannot be statically extracted

Import trace for requested module:
../wasm-sqlparser/cjs/out.cjs
../js-tableland/dist/cjs/helpers/parser.js
../js-tableland/dist/cjs/helpers/index.js
../js-tableland/dist/cjs/index.js
./db/api/db.ts
./db/api/auth.ts
./db/api/index.ts
./components/header.tsx
./app/layout.tsx
```
Let me know if either of you have thoughts about that @carsonfarmer and @joewagner. Thanks..

**Comment by asutula** [comment]
That error prints to my console whenever the parser is used by the sdk, but things seems to still work (the web app renders fine).

**Comment by joewagner** [comment]
> Let me know if either of you have thoughts about that @carsonfarmer and @joewagner. Thanks..

As a test I tried using this to run the SDK tests via `npm link`.  That results in a different error. 
```
/Users/jwagner/Workspaces/textile/github/wasm-sqlparser/cjs/out.cjs:595
export {};
^^^^^^

SyntaxError: Unexpected token 'export'
    at internalCompileFunction (node:internal/vm:73:18)
    at wrapSafe (node:internal/modules/cjs/loader:1149:20)
    at Module._compile (node:internal/modules/cjs/loader:1190:27)
    at Module.m._compile (/Users/jwagner/Workspaces/textile/github/js-tableland/node_modules/ts-node/src/index.ts:1618:23)
    at Module._extensions..js (node:internal/modules/cjs/loader:1280:10)
    at Object.require.extensions.<computed> [as .js] (/Users/jwagner/Workspaces/textile/github/js-tableland/node_modules/ts-node/src/index.ts:1621:12)
    at Module.load (node:internal/modules/cjs/loader:1089:32)
    at Function.Module._load (node:internal/modules/cjs/loader:930:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:169:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)
```
This is a little weird since line 595 does not exist in out.cjs.  If I change the outfile file extension to `.js` things work as expected.  Does changing the file extension fix your issue?

**Comment by asutula** [comment]
Looks like it causes a different error if I change `out.cjs` to `out.js` (along with other changes to the parser lib package.json to remain consistent with the new name):
```
- error ../js-tableland/dist/cjs/helpers/parser.js:4:20
Module not found: ESM packages (@tableland/sqlparser) need to be imported. Use 'import' to reference the package instead. https://nextjs.org/docs/messages/import-esm-externals
  2 | Object.defineProperty(exports, "__esModule", { value: true });
  3 | exports.validateTables = exports.validateTableName = exports.normalize = void 0;
> 4 | const sqlparser_1 = require("@tableland/sqlparser");
    |                    ^
  5 | const chains_js_1 = require("./chains.js");
  6 | async function normalize(sql, nameMap) {
  7 |     if (typeof sql !== "string") {

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
../js-tableland/dist/cjs/helpers/index.js
../js-tableland/dist/cjs/index.js
./db/api/db.ts
./db/api/auth.ts
./db/api/index.ts
./app/actions.ts
./node_modules/next/dist/build/webpack/loaders/next-flight-action-entry-loader.js?actions=%5B%5B%22%2FUsers%2Faaron%2FCode%2Ftextile%2Fstudio%2Fapp%2Factions.ts%22%2C%5B%22authenticated%22%2C%22nonce%22%2C%22login%22%2C%22register%22%5D%5D%5D&__client_imported__=true!
```

**Comment by carsonfarmer** [comment]
Just make sure you run fixup to create the package.json file in the cjs folder

**Comment by asutula** [comment]
Thanks @carsonfarmer, that was a problem indeed. I did all that, and used the `.js` extension, and still get:
```
../wasm-sqlparser/cjs/out.js
Critical dependency: require function is used in a way in which dependencies cannot be statically extracted

Import trace for requested module:
../wasm-sqlparser/cjs/out.js
../js-tableland/dist/cjs/helpers/parser.js
../js-tableland/dist/cjs/helpers/index.js
../js-tableland/dist/cjs/index.js
./db/api/db.ts
./db/api/auth.ts
./db/api/index.ts
./components/header.tsx
./app/layout.tsx
```

**Pull request merged**